openlibrary/plugins/openlibrary/js/dialog.js

```diff
@@ -6,4 +8,55 @@ export default function initDialogs() {
     $('.floaterShut').attr('href', 'javascript:;').on('click', () => $.fn.colorbox.close());
     // This will close the colorbox from the parent.
     $('.floaterShut--parent').on('click', () => parent.$.fn.colorbox.close());
+    [
+        ['.listClick', '#addList'],
+        ['#markdownPop', '#markdown' ],
+        ['#wikilink', '#wikicode'],
+        ['#coverPop,.coverPop', '#addImage'],
+        ['#coverPop', '#addCover'],
+        ['.coverLook', '#seeImage']
+    ].forEach(([ selector, href ]) => {
+        const $link = $(selector),
+            $dialog = $(href);
+
+        if ($link.length && $dialog.length) {
+            $link.on('click', () => {
+                if ($dialog.find('.tabs').length) {
+                    initTabs($dialog);
+                }
```

Comment:
I agree this is awkward. The problem is dialogs as currently rendered are rendered as hidden elements in the html.

If a jquery ui tabs element is in the DOM jquery ui will be loaded and the code run, but it will not be used until the dialog is open.

So what we really need is a way to delay this code execution until later.

This felt like the most clean way... essentially in the startup script we say enhance any tabs elements in the page but not inside a dialog.  Before the dialog opens we say enhance the tabs here. While we just have tabs here, in future maybe we need to run more code here.

I think the right solution is to pull the dialog html generation out of the template and have a specific dialog for each thing (e.g. this would be a ChangeCoverDialog).

Using data attributes here feels icky too as that complicates this journey.

Not really sure what to suggest here unless we just accept jquery ui is on the critical path which given it's size I'm keen to avoid.  I'll pull this out into a separate patch so as not to block progress and get the non controversial bits merged.

Reply:
This discussion has been punted down the road to 0b55fe8